enumerable_methods.rb

```diff
@@ -0,0 +1,65 @@
+module Enumerable
+  def my_each
+    return unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return unless block_given?
+
+    index = 0
+    for i in self
+      yield i, index
+      index += 1
+    end
+  end
+
+  def my_select
+    arr = []
+    my_each { |n| arr.push(n) if yield n } if block_given?
+    arr
+  end
+
+  def my_all?
+    my_select { |n| yield n }.length == length if block_given?
+  end
+
+  def my_any?
+    my_select { |n| yield n }.length.positive? if block_given?
+  end
+
+  def my_none?
+    !my_any? if block_given?
+  end
+
+  def my_count
+    my_select { |n| yield n }.length if block_given?
+  end
```

Comment:
- [ ] enumerables #my_count returns the number of items in enum through enumeration
      Failure/Error: expect(array.my_count).to eq array.count

- [ ] enumerables #my_count returns the number of items in enum through enumeration::range
      Failure/Error: expect(range.my_count).to eq range.count

- [ ] enumerables #my_count counts the number of items in enum that are equal to item if an argument is given
      Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)